cobalt/updater/configurator.cc

```diff
@@ -0,0 +1,422 @@
+  } else {
+    SetChannel(persisted_channel);
+  }
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+          switches::kUseUncompressedUpdates)) {
+    use_compressed_updates_.store(false);
+  } else {
+    use_compressed_updates_.store(true);
+  }
+}
+Configurator::~Configurator() {
+  LOG(INFO) << "Configurator::~Configurator";
+}
+
+base::TimeDelta Configurator::InitialDelay() const {
+  return base::Seconds(0);
+}
+
+base::TimeDelta Configurator::NextCheckDelay() const {
+  return base::Hours(5);
+}
+
+base::TimeDelta Configurator::OnDemandDelay() const {
+  return base::Seconds(0);
+}
+
+base::TimeDelta Configurator::UpdateDelay() const {
+  return base::Seconds(0);
+}
+
+std::vector<GURL> Configurator::UpdateUrl() const {
+#if !defined(COBALT_BUILD_TYPE_GOLD)
+  if (allow_self_signed_packages_ && !update_server_url_.empty()) {
+    return std::vector<GURL>{GURL(update_server_url_)};
+  }
+#endif  // !defined(COBALT_BUILD_TYPE_GOLD)
+  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
+          switches::kUseQAUpdateServer)) {
+    return std::vector<GURL>{GURL(kUpdaterJSONDefaultUrlQA)};
+  } else {
+    return std::vector<GURL>{GURL(kUpdaterJSONDefaultUrl)};
+  }
+}
+
+std::vector<GURL> Configurator::PingUrl() const {
+  return UpdateUrl();
+}
+
+std::string Configurator::GetProdId() const {
+  return "cobalt";
+}
+
+base::Version Configurator::GetBrowserVersion() const {
+  std::string version(COBALT_VERSION);
+  // base::Version only accepts numeric versions. Return Cobalt major version.
+  int first_dot = version.find_first_of(".");
+  if (first_dot != std::string::npos) {
+    return base::Version(version.substr(0, first_dot));
+  }
+  return base::Version("");
+}
+
+std::string Configurator::GetBrand() const {
+  return {};
+}
+
+std::string Configurator::GetLang() const {
+  const char* locale_id = SbSystemGetLocaleId();
+  if (!locale_id) {
+    // If Starboard failed to determine the locale, return a hardcoded, but
+    // valid BCP 47 language tag as required by
+    // https://html.spec.whatwg.org/commit-snapshots/e2f08b4e56d9a098038fb16c7ff6bb820a57cfab/#language-preferences
+    return "en-US";
+  }
+  std::string locale_string(locale_id);
+  // POSIX platforms put time zone id at the end of the locale id, like
+  // |en_US.UTF8|. We remove the time zone id.
+  int first_dot = locale_string.find_first_of(".");
+  if (first_dot != std::string::npos) {
+    return locale_string.substr(0, first_dot);
+  }
+  return locale_string;
+}
+
+std::string Configurator::GetOSLongName() const {
+  return GetDeviceProperty(kSbSystemPropertyPlatformName);
+}
+
+base::flat_map<std::string, std::string> Configurator::ExtraRequestParams()
+    const {
+  base::flat_map<std::string, std::string> params;
+  params.insert(std::make_pair("SABI", SB_SABI_JSON_ID));
+  params.insert(std::make_pair("sbversion", std::to_string(SB_API_VERSION)));
+
+  // The flag name to force an update is changed to "is_forced_update_" to cover
```

Comment:
nit: Escape variables in comments either with backticks or pipe
symbols (`|`). Code will be indexed that way. Apply throughout.